looper/cli_pydantic.py

```diff
@@ -0,0 +1,143 @@
+"""
+CLI script using `pydantic-argparse` for parsing of arguments
+
+Arguments / commands are defined in `command_models/` and are given, eventually, as
+`pydantic` models, allowing for type-checking and validation of arguments.
+
+Note: this is only a test script so far, and coexists next to the current CLI
+(`cli_looper.py`), which uses `argparse` directly. The goal is to eventually
+replace the current CLI with a CLI based on above-mentioned `pydantic` models,
+but whether this will happen with `pydantic-argparse` or another, possibly self-
+written library is not yet clear.
+It is well possible that this script will be removed again.
+"""
+
+import os
+import sys
+
+import pydantic_argparse
+import yaml
+from pephubclient import PEPHubClient
+
+from divvy import select_divvy_config
+
+from . import __version__
+from .cli_looper import _proc_resources_spec
+from .command_models.commands import TopLevelParser
+from .const import *
+from .divvy import DEFAULT_COMPUTE_RESOURCES_NAME, select_divvy_config
+from .exceptions import *
+from .looper import *
+from .parser_types import *
+from .project import Project, ProjectContext
+from .utils import (
+    dotfile_path,
+    enrich_args_via_cfg,
+    is_registry_path,
+    read_looper_dotfile,
+)
+
+
+def main() -> None:
+    parser = pydantic_argparse.ArgumentParser(
+        model=TopLevelParser,
+        prog="looper",
+        description="pydantic-argparse demo",
+        add_help=True,
+    )
+    args = parser.parse_typed_args()
+    print(args)
+    print("#########################################")
+
+    # here comes adapted `cli_looper.py` code
+    looper_cfg_path = os.path.relpath(dotfile_path(), start=os.curdir)
+    try:
+        looper_config_dict = read_looper_dotfile()
+
+        for looper_config_key, looper_config_item in looper_config_dict.items():
+            print(looper_config_key, looper_config_item)
+            setattr(args, looper_config_key, looper_config_item)
+
+    except OSError:
+        parser.print_help(sys.stderr)
+        raise ValueError(
+            f"Looper config file does not exist. Use looper init to create one at {looper_cfg_path}."
+        )
```

Comment:
It seems here (from the `ValueError`'s message) like the assumption is that the hypothetical `OSError` arises only if the config file doesn't exist? If so, the try/except block should reflect the narrowness of that scenario, b/c at first glance it looks like anything in the loop _could_ cause the exception. 

Also the definition of `looper_cfg_path` could be moved closer to its (sole) use site, so that it's clearer how that variable's being used. 

Finally, it seems this code assumes that `read_looper_dotfile` will try to read a path that coincides with what's stored in `looper_cfg_path`, but there's nothing in the implementation that enforces that assumption; what if the implementation of `read_looper_dotfile` changed such that it was reading from a different path? It'd be nice to have the error message automatically adapted to that change rather than needing to do it manually here.

Reply:
I agree with all your points and I think it would be worth addressing them. But I'm not sure that file / PR is necessary the right place: as the module docstring says, this is mostly a test script, and a testimony to this is that the code you point out here and in other comments below is literally copy & pasted from `looper/cli_looper.py`. So I think a separate PR to address the issues you mention in `cli_looper.py` would be ideal.
That being said, we of course would have been better off had we refactored `looper/cli_looper.py` to make that code reusable right from the start, so we have to change it in only once place. But then again, the copy & pasted code is also slightly adapted to make it work with our `pydantic`(`-argparse`)-based argument definition.

Reply:
makes sense separate PRs yeah, since then related tweaks could be made to both modules where implementations are kept separate rather than shared, or in a single place that then both are pull from.